### ISSUE TITLE
vqa_token_re_layoutlm_postprocess error

### ISSUE THREAD

**Issue by harshsummit** [PaddlePaddle/PaddleOCR#8995]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：Ubuntu 20.04
- 版本号/Version：Paddle：2.4.1  PaddleOCR：2.6 release 问题相关组件/Related components：
- 运行指令/Command Code：python tools/train.py -c configs/kie/vi_layoutxlm/re_layoutxlm_funsd.yml
- 完整报错/Complete Error Message：TypeError: list indices must be integers or slices, not tuple

<img width="668" alt="image" src="https://user-images.githubusercontent.com/122161388/217217136-45b3fbd5-e3a1-4248-8065-adc79d841af7.png">

error is in the following piece of code in vqa_token_re_layoutlm_postprocess.py file which occurs during Postprocess
<img width="451" alt="image" src="https://user-images.githubusercontent.com/122161388/217217280-4f9bd1ac-9532-45ed-9bb9-85953ae8f107.png">

**Comment by harshsummit** [comment]
the yml config use is

`
Global:
  use_gpu: False
  epoch_num: &epoch_num 200
  log_smooth_window: 10
  print_batch_step: 10
  save_model_dir: ./output/re_layoutxlm_funsd
  save_epoch_step: 2000
  # evaluation is run every 10 iterations after the 0th iteration
  eval_batch_step: [ 0, 19 ]
  cal_metric_during_train: False
  save_inference_dir:
  use_visualdl: False
  seed: 2022
  infer_img: train_data/FUNSD/testing_data/images/83624198.png
  save_res_path: ./output/re_layoutxlm_funsd/res/

Architecture:
  model_type: kie
  algorithm: &algorithm "LayoutXLM"
  Transform:
  Backbone:
    name: LayoutXLMForRe
    pretrained: True
    checkpoints:

Loss:
  name: LossFromOutput
  key: loss
  reduction: mean

Optimizer:
  name: AdamW
  beta1: 0.9
  beta2: 0.999
  clip_norm: 10
  lr:
    learning_rate: 0.00005
    warmup_epoch: 10
  regularizer:
    name: L2
    factor: 0.00000
    
PostProcess:
  name: VQAReTokenLayoutLMPostProcess

Metric:
  name: VQAReTokenMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/FUNSD/training_data/images/
    label_file_list: 
      - ./train_data/FUNSD/train.json
      # - ./train_data/FUNSD/train.json
    ratio_list: [ 1.0 ]
    transforms:
      - DecodeImage: # load image
          img_mode: RGB
          channel_first: False
      - VQATokenLabelEncode: # Class handling label
          contains_re: True
          algorithm: *algorithm
          class_path: &class_path ./train_data/FUNSD/class_list.txt
          use_textline_bbox_info: &use_textline_bbox_info True
      - VQATokenPad:
          max_seq_len: &max_seq_len 512
          return_attention_mask: True
      - VQAReTokenRelation:
      - VQAReTokenChunk:
          max_seq_len: *max_seq_len
      - Resize:
          size: [224,224]
      - NormalizeImage:
          scale: 1
          mean: [ 123.675, 116.28, 103.53 ]
          std: [ 58.395, 57.12, 57.375 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          # dataloader will return list in this order
          keep_keys: [ 'input_ids', 'bbox', 'attention_mask', 'token_type_ids', 'image', 'entities', 'relations']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 2
    num_workers: 4
    collate_fn: ListCollator

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/FUNSD/testing_data/images/
    label_file_list: 
      - ./train_data/FUNSD/test.json
      # - ./train_data/FUNSD/test.json
    transforms:
      - DecodeImage: # load image
          img_mode: RGB
          channel_first: False
      - VQATokenLabelEncode: # Class handling label
          contains_re: True
          algorithm: *algorithm
          class_path: ./train_data/FUNSD/class_list.txt
          use_textline_bbox_info: *use_textline_bbox_info
      - VQATokenPad:
          max_seq_len: *max_seq_len
          return_attention_mask: True
      - VQAReTokenRelation:
      - VQAReTokenChunk:
          max_seq_len: *max_seq_len
      - Resize:
          size: [224,224]
      - NormalizeImage:
          scale: 1
          mean: [ 123.675, 116.28, 103.53 ]
          std: [ 58.395, 57.12, 57.375 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          # dataloader will return list in this order
          keep_keys: [ 'input_ids', 'bbox', 'attention_mask', 'token_type_ids', 'image', 'entities', 'relations']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 8
    num_workers: 4
    collate_fn: ListCollator
`

**Comment by andyjpaddle** [comment]
please try to update paddlenlp package

**Comment by harshsummit** [comment]
> please try to update paddlenlp package

Tried that too, the problem is with the line 61

**Comment by andyjpaddle** [comment]
which version did you use?

**Comment by harshsummit** [comment]
tried 2.3.0, 2.5.0 and 2.4.0 as well

**Comment by andyjpaddle** [comment]
try 2.3.1?

**Comment by harshsummit** [comment]
doesn't seem to be error for paddlenlp, if u see the error is for accessing the pred_relation variable which is a numpy array, 
and to access it we are accessing it as pred_relation[0,0,0] + 1, this doesnt give a required output for slicing in python

**Comment by andyjpaddle** [comment]
I tried 2.4.4 and it works well, layoutxml depends on the paddlenlp

**Comment by harshsummit** [comment]
Tried, its error with numpy

**Comment by harshsummit** [comment]
the array accessing works at 1.21.5 but it gives error for polygon dependency as it requires 1.23.5

**Comment by harshsummit** [comment]
ig we need new code for the file depending upon new numpy versions

**Comment by harshsummit** [comment]
@andyjpaddle do you have any idea what does this .numpy() function do
<img width="328" alt="image" src="https://user-images.githubusercontent.com/122161388/217464486-d1895d51-5e31-4263-a148-104ba17e136d.png">

**Comment by andyjpaddle** [comment]
> what does this .numpy() function do

this function is convert the tensor into a numpy array, so it can be used in subsequent calculation

**Comment by harshsummit** [comment]
> > what does this .numpy() function do
> 
> this function is convert the tensor into a numpy array, so it can be used in subsequent calculation
@andyjpaddle 
I checked logging the type of preds is of type `dict`, and preds_relation is of type `list` and its empty -> [[] [] [] [] []], 
therefore it is not able to detect it as tensor or convert it into numpy

**Comment by andyjpaddle** [comment]
did you test the official model? i tested it and it works well

**Comment by harshsummit** [comment]
official model is on xfund dataset, i was trying for FUNSD dataset

**Comment by harshsummit** [comment]
Do you have any pre trained FUNSD KIE model available?

**Comment by WenmuZhou** [comment]
we only release the model trained on xfund.
for FUNSD, does you convert the dataset to xfund format?

**Comment by harshsummit** [comment]
yes i did

**Comment by WenmuZhou** [comment]
You can try paddlenlp>=2.4.1, in versions after 2.4.1, we have made relatively large changes to the code to support dynamic to static

**Comment by harshsummit** [comment]
Tried that too @WenmuZhou

**Comment by harshsummit** [comment]
@WenmuZhou @andyjpaddle on compiling PaddleNLP to latest version, It gives Floating Point Exception (Core Dumped)

**Comment by harshsummit** [comment]
I really think there is some problem in training the KIE models with FUNSD dataset, please look into it, there are already open issues regarding FUNSD dataset from past 1 years which are unresolved.

I dont think anybody has till date sucessfully trained kie model with FUNSD dataset and then used it for inference

**Comment by harshsummit** [comment]
@andyjpaddle @WenmuZhou any update???

**Comment by NaumanHSA** [comment]
@andyjpaddle @WenmuZhou Have you guys any update regarding training KIE on FUNSD dataset? I'm facing the "Core Dumped" exception. Also, can you guys tell me how much GPU memory is required for KIE training?

**Comment by harshsummit** [comment]
Same here @NaumanHSA , there is someproblem with code, Ive been trying since 2 weeks for the same

**Comment by NaumanHSA** [comment]
@harshsummit I tried training KEI model on XFUND dataset, and it started normally. I think there's an issue with the FUNSD dataset but I don't know how to figure it out? About the memory thing, with desktop RTX 4070 ti, CUDA 11.7, cuDNN 8.4, Ubuntu 20.04, I was able to start KIE training on XFUND dataset.

**Comment by harshsummit** [comment]
@NaumanHSA exactly same, tried for XFUND works, but for FUNSD throws error

**Comment by NaumanHSA** [comment]
@harshsummit I tried on a custom dataset with a very few images, it started as normal. Then I verified the FUNSD dataset by just reading the labels, rendering the boxes and text using opencv, to check if there could be some incorrect box annotations but all of the images were labeled correctly. 

In my second test, I found that some special characters in the dataset were encoded as utf8. I thought that might be an issue (was not sure but gave it a try to remove all those characters from the dataset). Still got the same error. I think this problem is with the paddleNLP package.
